bin/smash

```diff
@@ -225,10 +220,11 @@ def main():
          
             log_quiet('running %s in %s as %s testset #%s %s',t,config.rundir,config.oc_account_name,i,repr(ts)) # log_quiet
 
-         reporter.testcase_start(barename(t),j,i,smashbox.no_engine)
-
          if not args.dry_run:
-            run_multiprocessing_engine(config,os.path.abspath(t),reporter)
+            setattr(config, "loop_i",j) 
+            setattr(config, "testset_i",i) 
+            setattr(config, "test_doc",smashbox.no_engine.__doc__) 
+            run_multiprocessing_engine(config,os.path.abspath(t))
 
```

Comment:
now config in each iteration will modify loop_i, testset_i and test_doc, which will be needed in the reporter. . It is reducing the number of arguments passed to next functions.